modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/Filter.js

```diff
@@ -14,20 +14,27 @@ import DateRangeFilter, {
 	getOdataString as getDateRangeFilterOdataString,
 	getSelectedItemsLabel as getDateRangeFilterSelectedItemsLabel,
 } from './DateRangeFilter';
+import DynamicFilter, {
+	getOdataString as getDynamicFilterOdataString,
+	getSelectedItemsLabel as getDynamicFilterSelectedItemsLabel,
+} from './DynamicFilter';
 import SelectionFilter, {
 	getOdataString as getSelectionFilterOdataString,
 	getSelectedItemsLabel as getSelectionFilterSelectedItemsLabel,
 } from './SelectionFilter';
 
 const FILTER_TYPE_COMPONENT = {
 	dateRange: DateRangeFilter,
+	dynamic: DynamicFilter,
```

Comment:
Dynamic filter is a subtype of `SelectionFilter`. It should be that exact filter, just with different data source (picklist and not `items` given in a prop). Unless I'm missing something big, we shoudn't need a new component. There is a bunch of logic in `SelectionFilter` that is missing from this solution. Can we reuse it?

Reply:
Oops, didn't pay attention to that existing filter. Maybe we can even send the items from the server, instead of the URL: since they are immutable, there's no need to retrieve them from the browser... What do you think? :thinking:

Reply:
Why not both? ™

Sending the items from server is good for performance, no need for initial delay. Providing URL enables the autocomplete feature alter on.

Reply:
existing selection filter should work here. Object feature uses Selection filter for a picklist, just providing the picklist item fetch url based on `list-type-definition-id`. We can improve it however for this task I'd recommend configuring the Selection filter

Reply:
OK, so I've tried three ways to configure the filter editor:

1. Send the items from the server, no apiURL
2. Send the items from the server plus the apiURL
3. Send only the apiURL

1 seems great as long as the picklists are not huge. This is because it's more performant and offers a better UI because we don't need any trip to the server. It has the downside that there's no autocompletion, but the truth is that I haven't been able to make autocomplete work for the other cases, so I wonder if it is really needed for picklists. As a user I would expect picklist items to show completely in the UI, not having to guess what are the possible values.

2 works like 1. I can't see any difference even with 12 items in the picklist.

3 works like 1 too, but there's a trip to the server to get the items and, additionally, the spinner is shown as if it was autocompleting, when I search. However, nothing happens, since the items already appear at the bottom.

So, my question is: what do I do? If I had to choose I would go with 1 for now...

Reply:
For the first version, (1) seems fine. For (2) to fully make sense, we would probably want an option in DSM to toggle autocomplete on and off.

Reply:
after some deeper dive we saw that API url is a kind of mutually exclusive with respect to items. So we can basically use 1 or 3. For me 1 is fine though we need to be flexible in case we find some issue in the near future

Reply:
I've force pushed solution 1 for now.

Reply:
@markocikos Can you please make a quick review to see if the new commits make sense so that we can forward it to Brian :pray: ? Thx.